pkg/azure/provider/provider.go

```diff
@@ -0,0 +1,200 @@
+	if err != nil {
+		err = status.Error(codes.Internal, fmt.Sprintf("failed to get virtual machine for VM: [resourceGroup: %s, name: %s], Err: %v", resourceGroup, vmName, err))
+		return
+	}
+	/*
+		It is possible to have left over NIC's and Disks even if the VM is no longer there. This is made possible because in the earlier version of this provider
+		implementation the cascade-delete is not enabled for NICs and Disks on deletion of the VM. Thus, it's possible that while the VM gets deleted the NIC's and Disks are left behind.
+		Once all the VirtualMachines are launched with cascade-delete enabled for NICs and Disks then this can be removed.
+	*/
+	if vm == nil {
+		klog.Infof("VirtualMachine [resourceGroup: %s, name: %s] does not exist. Skipping deletion of VirtualMachine", providerSpec.ResourceGroup, vmName)
+		// check if there are leftover NICs and Disks that needs to be deleted.
+		if err = helpers.CheckAndDeleteLeftoverNICsAndDisks(ctx, d.factory, vmName, connectConfig, providerSpec); err != nil {
+			return
+		}
+	} else {
+		err = helpers.UpdateCascadeDeleteOptionsAndDeleteVM(ctx, vmAccess, resourceGroup, vm)
+		if err != nil {
+			return
+		}
+		klog.Infof("Successfully deleted all Machine resources[VM, NIC, Disks] for [ResourceGroup: %s, VMName: %s]", providerSpec.ResourceGroup, vmName)
+	}
+	resp = &driver.DeleteMachineResponse{}
+	return
+}
+
+func (d defaultDriver) GetMachineStatus(ctx context.Context, req *driver.GetMachineStatusRequest) (resp *driver.GetMachineStatusResponse, err error) {
+	defer instrument.RecordDriverAPIMetric(err, getMachineStatusOperationLabel, time.Now())
+	providerSpec, connectConfig, err := helpers.ExtractProviderSpecAndConnectConfig(req.MachineClass, req.Secret)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceGroup := providerSpec.ResourceGroup
+	vmName := req.Machine.Name
+	vmAccess, err := d.factory.GetVirtualMachinesAccess(connectConfig)
+	if err != nil {
+		err = status.Error(codes.Internal, fmt.Sprintf("Failed to create virtual machine access to process request: [ResourceGroup: %s, VMName: %s], Err: %v", resourceGroup, vmName, err))
+		return
+	}
+
+	// After getting response for Query: [https://github.com/Azure/azure-sdk-for-go/issues/21031] replace this call with a more optimized variant to check if a VM exists.
+	vm, err := clienthelpers.GetVirtualMachine(ctx, vmAccess, resourceGroup, vmName)
+	if err != nil {
+		err = status.Error(codes.Internal, fmt.Sprintf("Failed to get VM: [ResourceGroup: %s, Name: %s], Err: %v", resourceGroup, vmName, err))
+		return
+	}
+	if vm == nil {
+		err = status.Error(codes.NotFound, fmt.Sprintf("VM: [ResourceGroup: %s, Name: %s] is not found", resourceGroup, vmName))
+		return
+	}
+	// Enhance the response as proposed in [https://github.com/gardener/machine-controller-manager-provider-azure/issues/88] once that is taken up.
+	resp = helpers.ConstructGetMachineStatusResponse(providerSpec.Location, vmName)
```

Comment:
A log to specify that VM was found would be helpful here.

Reply:
sure, added